components/collapsible-panel/collapsible-panel-group.js

```diff
@@ -0,0 +1,26 @@
+import { css, html, LitElement } from 'lit';
+import { SkeletonMixin } from '../skeleton/skeleton-mixin.js';
+
+/**
+ * A component that renders collapsible panels
+ * @slot default - Slot for panels. Only accepts `d2l-collapsible-panel`
+ */
+class CollapsiblePanelGroup extends SkeletonMixin(LitElement) {
+
+	static get styles() {
+		return [super.styles, css`
+			:host ::slotted(*) {
+				display: none;
+			}
+			:host ::slotted(d2l-collapsible-panel) {
+				display: unset;
+			}
+		`];
+	}
+
+	render() {
+		return html`<slot></slot>`;
```

Comment:
The group component can perhaps handle the panel spacing automatically using flex:

```html
<div style="display: flex; flex-direction: column; gap: 0.5rem;">
    <slot></slot>
</div>
```

Some inspection of the contents (via `slotchange` or other means) would still be necessary to know if the children are inline.

I know you're also trying to do some stuff with the border though.